packages/web/src/components/premium-content-purchase-modal/PremiumContentPurchaseModal.tsx

```diff
@@ -275,6 +276,8 @@ export const PremiumContentPurchaseModal = () => {
     dispatch(cleanupUSDCRecovery())
   }, [onClosed, dispatch])
 
+  useManagedAccountNotAllowedCallback(isOpen, handleClose)
```

Comment:
I'm curious, since this uses `useEffect`, if the modal actually starts showing/initializing before we navigate away? Might be nice to prevent passing `isOpen` to the modal at all.

Reply:
it doesn't show at all actually. we can revisit though if it proves to be weird during QA?

Reply:
Oh okay great, maybe since it runs before whatever effect would happen in the modal, we're good.